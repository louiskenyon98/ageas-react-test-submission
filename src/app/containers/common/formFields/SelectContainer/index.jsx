import React from 'react';
import {connect} from 'react-redux';
import Select from '../../../../components/common/formFields/Select';
import {filterOptions} from '../../../../config/filterOptions';

import {onChangeFilterValue} from '../../../../actions/filterValues/index.actions';


class SelectContainer extends React.Component {
    render() {
        return (
            <Select
                options={filterOptions}
                value={this.props.filterOption}
                onChange={this.props.onChangeFilterValue}
            />
        )
    }
}

const mapStateToProps = (state) => {
    return {
        filterOption: state.notes.filterOption
    }
};
export default connect(mapStateToProps, {onChangeFilterValue})(SelectContainer);
