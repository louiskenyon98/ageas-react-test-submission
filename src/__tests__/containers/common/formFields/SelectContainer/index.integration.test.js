import React from 'react';
import {mount} from 'enzyme';
import thunk from 'redux-thunk';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store'
import SelectContainer from '../../../../../app/containers/common/formFields/SelectContainer/index';

describe('SelectContainer Integration Test', () => {
    let props = {};
    let store = {};
    let mountedComponent;
    const middleware = [thunk];
    const mockStore = configureMockStore(middleware);
    let wrapper = () => {
        if (!mountedComponent) {
            mountedComponent = mount(
                <Provider store={store}>
                    <SelectContainer {...props}/>
                </Provider>
            )
        }
        return mountedComponent;
    };
    beforeEach(() => {
        props = {};
        store = mockStore({
            notes: {
                filterOption: "createdAt"
            }
        });
        mountedComponent = undefined;
    });
    it('should render Select component with options, a value and an onChange function', () => {
        expect(wrapper()).toMatchSnapshot();
    });
    it('should change value from createdAt to updatedAt', () => {
        expect(wrapper().find('select').first().props().value).toBe("createdAt");
        store = mockStore({
            notes: {
                filterOption: "updatedAt"
            }
        });
        wrapper().setProps({store});
        wrapper().update();
        expect(wrapper().find('select').first().props().value).toBe("updatedAt");
    });
    it('should dispatch ONCHANGE_FILTER_VALUE with updatedAt', () => {
        expect(store.getActions().length).toBe(0);
        wrapper().find('select').first().props().onChange({
            target: {
                value: 'updatedAt'
            }
        });
        expect(store.getActions().length).toBe(1);
        expect(store.getActions()[0]).toEqual({
            payload: "updatedAt",
            type: "ONCHANGE_FILTER_VALUE"
        });
    });
});
