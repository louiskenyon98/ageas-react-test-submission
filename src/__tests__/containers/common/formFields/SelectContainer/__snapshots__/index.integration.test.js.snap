// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectContainer Integration Test should render Select component with options, a value and an onChange function 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(SelectContainer)>
    <SelectContainer
      filterOption="createdAt"
      onChangeFilterValue={[Function]}
    >
      <Select
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "Last created",
              "value": "createdAt",
            },
            Object {
              "label": "Last edited",
              "value": "updatedAt",
            },
          ]
        }
        value="createdAt"
      >
        <select
          onChange={[Function]}
          value="createdAt"
        >
          <option
            key="createdAt"
            value="createdAt"
          >
            Last created
          </option>
          <option
            key="updatedAt"
            value="updatedAt"
          >
            Last edited
          </option>
        </select>
      </Select>
    </SelectContainer>
  </Connect(SelectContainer)>
</Provider>
`;
